### ISSUE TITLE
Rules don't work with remote execution

### ISSUE THREAD

**Issue by purkhusid** [bazelbuild/rules_dotnet#221]
I've been testing the rules with remote execution via https://buildbuddy.io and it seems like they do not work.

This can be reproduced in this repo https://github.com/purkhusid/bazel-bootstrap by following the README.md and and then run `bazel build //...  --config=remote`

I've been trying to debug it but I've got little experience debugging remote execution. I can work on a fix if I could get some guidance on debugging remote execution.

**Comment by purkhusid** [comment]
The exact error I get is:

```
(00:31:18) ERROR: /workspace/bazel-bootstrap/src/proto/echo/models/BUILD.bazel:54:21: Compiling src/proto/echo/models:csharp.dll failed: (Stopped (tty output)): dotnet failed: error executing command 
  (cd /workspace/bazel_output_base/execroot/monorepo && \
  exec env - \
  external/core_sdk_linux_amd64_3.1.100/core/dotnet external/core_sdk_linux_amd64_3.1.100/core/sdk/3.1.100/Roslyn/bincore/csc.dll /noconfig @bazel-out/k8-fastbuild/bin/src/proto/echo/models/csharp.dll.param)
Execution platform: //:buildbuddy_platform
EhUI0gEQ/uSEEhivlxsgByjbAzDR8yBCgwEKCE9qUDZUZDI1EgsI096hgQYQ+cbJERoLCNPeoYEGEPyAnh4iCwjV3qGBBhDs7rRsKgsI096hgQYQiIueHjIMCNPeoYEGEMGS4PIBOgwI096hgQYQ7tHn8wFCDAjU3qGBBhDqqpbIAkoMCNTeoYEGEP+1lsgCUgsI1d6hgQYQvOi0bA==
It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '3.1.0' was not found.
  - No frameworks were found.

You can resolve the problem by installing the specified framework and/or SDK.

The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=3.1.0&arch=x64&rid=ubuntu.16.04-x64
```

**Comment by purkhusid** [comment]
I also had to create a custom platform for the remote exec as seen here: https://github.com/purkhusid/bazel-bootstrap/blob/3db9e4977e4d8df7f3cf8482ab5c00a5264673a3/BUILD.bazel#L27

**Comment by purkhusid** [comment]
Ok, I've been digging into this and by adding the `-d` argument here: https://github.com/bazelbuild/rules_dotnet/blob/312dfb42f1e55dd7cbc5da0d1958de302b9643d1/dotnet/private/actions/assembly_core.bzl#L181
it starts showing the missing files.

It seems like most of the `.so` `.dylib` `.dll` from `sdk/shared/...` are missing when I use remote execution. 
These files are all declared in the `<version>-runtime.bzl` but they don't seem to be part of the compilation action.

**Comment by purkhusid** [comment]
@tomaszstrejczek If I change this line: https://github.com/bazelbuild/rules_dotnet/blob/312dfb42f1e55dd7cbc5da0d1958de302b9643d1/dotnet/private/BUILD.core.bazel#L49

To `data = glob(["core/**/*"]),`

everything works with remote execution. This is not really an acceptable solution though since it pulls the whole SDK into each action.

**Comment by purkhusid** [comment]
@tomaszstrejczek  Proposed fix in #224

**Comment by purkhusid** [comment]
Fixed in #224
